docs/tui.md

```diff
@@ -0,0 +1,234 @@
+- Degrades gracefully to simpler layouts for small terminals
+- Maintains full-width consistency for all panels
+
+### Keyboard Navigation
+
+- Tab/Shift+Tab to navigate between panels
+- ESC to exit overlays
+- Shortcut keys for common operations
+
+### Port Forward Management
+
+- Monitor active port forwards (Prometheus, Grafana, Alloy Metrics) with status indicators.
+- Prometheus (MC) and Grafana (MC) are standard and always use the Management Cluster context.
+- Alloy Metrics port-forwarding follows this logic:
+  - If both a Management Cluster and a Workload Cluster are configured, Alloy Metrics connects to the Workload Cluster.
+  - If only a Management Cluster is configured, Alloy Metrics connects to that Management Cluster.
+- Restart individual port forwards when needed using the 'r' key with the panel focused.
+
+### Dark Mode Support
+
+- Complete dark mode support with 'D' key toggle
+- Adaptive colors for all UI elements
+- Proper contrast in both modes for readability
+
+### Focus System
+
+- Focused panels have distinct visual styling
+- Focus can be moved between all interactive elements
+- Current focus affects context-specific operations (e.g., 'r' to restart focused port forward)
+
+### Overlays
+
+- Help overlay ('h') displays all keyboard shortcuts
+- Log overlay ('L') for expanded log viewing when screen space is limited
+
+## Implementation Details
+
+### Port Forwarding Management
+
+Port forwards are managed by:
+- `portforward_handlers.go`: Logic for setting up, monitoring, and restarting port forwards, with specific behavior for each service:
+  - Prometheus (MC) and Grafana (MC) always connect to the Management Cluster
+  - Alloy Metrics connects to the Workload Cluster if one is specified, otherwise it connects to the Management Cluster
+- `portForwardProcess` struct: Tracks process state (including which cluster it targets), output, and errors for each port-forward.
+- Status update messages: Keep the UI in sync with actual process status for all services.
+
+### Context Switching
+
+The TUI handles context switching through:
+- `connection_flow.go`: Functions to manage the connection flow
+- `handlers.go`: Event handlers for keyboard shortcuts
+- Asynchronous operations to update the UI as contexts change
+
+### Viewport Management
+
+Scrollable log views are implemented using Bubble Tea's viewport component:
+- `mainLogViewport`: For the in-line log panel
+- `logViewport`: For the expandable log overlay
+- Mouse wheel scrolling support in both viewports
+
+### Debug Features
+
+The TUI includes debugging capabilities:
+- Toggle debug mode with 'z' key
+- View raw dimensions and layout calculations
+- Detect and display color profile information
+
+## File Structure
+
+- `model.go`: Core model definition, Update() and View() methods
+- `view_helpers.go`: Helper functions for rendering UI components
+- `handlers.go`: Key press and event handlers
+- `styles.go`: UI styling definitions
```

Comment:
This feature is not implemented in the code, so we should remove this line from the documentation.

```suggestion
```